activemodel/lib/active_model/type/registry.rb

```diff
@@ -9,15 +9,16 @@ def initialize
       end
 
       def register(type_name, klass = nil, **options, &block)
-        block ||= proc { |_, *args| klass.new(*args) }
+        block ||= proc { |_, *args, **kwargs| klass.new(*args, **kwargs) }
```

Comment:
> We can't use **kwargs in this proc since the registered klass.new might not allow empty hash (e.g. Integer.new).
https://buildkite.com/rails/rails/builds/65803#d383432f-7419-4a27-91a9-6843aac234f1/934-942

Still failed due to `**kwargs`.
https://buildkite.com/rails/rails/builds/65809#f86218b7-b104-40b1-bea5-2d305d78b2ed/996-1004

Reply:
Sorry, that was a squash mistake. I fixed it now.